ksp_plugin/pile_up.cpp

```diff
@@ -190,6 +192,65 @@ void PileUp::NudgeParts() const {
   }
 }
 
+void PileUp::WriteToMessage(not_null<serialization::PileUp*> message) const {
+  for (auto const part : parts_) {
+    message->add_part_id(part->part_id());
+  }
+  mass_.WriteToMessage(message->mutable_mass());
+  intrinsic_force_.WriteToMessage(message->mutable_intrinsic_force());
+  psychohistory_->WriteToMessage(message->mutable_psychohistory(),
+                                 /*forks=*/{});
+  for (auto const& pair : actual_part_degrees_of_freedom_) {
+    auto const part = pair.first;
+    auto const& degrees_of_freedom = pair.second;
+    degrees_of_freedom.WriteToMessage(&(
+        (*message->mutable_actual_part_degrees_of_freedom())[part->part_id()]));
+  }
+  for (auto const& pair : apparent_part_degrees_of_freedom_) {
+    auto const part = pair.first;
+    auto const& degrees_of_freedom = pair.second;
+    degrees_of_freedom.WriteToMessage(&(
+        (*message
+              ->mutable_apparent_part_degrees_of_freedom())[part->part_id()]));
+  }
+}
+
+not_null<std::unique_ptr<PileUp>> PileUp::ReadFromMessage(
+    serialization::PileUp const& message,
+    std::function<not_null<Part*>(PartId)> const& part_id_to_part) {
```

Comment:
I'm not sure how you're going to deserialize the pile ups consisting of a dummy part...